app/src/androidTest/java/ch/epfl/sdp/CreateEventFragmentTest.java

```diff
@@ -0,0 +1,125 @@
+package ch.epfl.sdp;
+
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.runner.AndroidJUnit4;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import ch.epfl.sdp.db.Database;
+import ch.epfl.sdp.ui.event.CreateEventFragment;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.clearText;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.matcher.RootMatchers.withDecorView;
+import static androidx.test.espresso.matcher.ViewMatchers.withHint;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static java.lang.Thread.sleep;
+import static org.hamcrest.Matchers.is;
+
+@RunWith(AndroidJUnit4.class)
+public class CreateEventFragmentTest {
+
+    private Database mDb = new MockDatabase();
+    private static MockEvents mMockEvents = new MockEvents();
+
+    private static final String DATE = "20/04/2020";
+    private static final String TITLE = mMockEvents.getNextEvent().getTitle();
+    private static final String DESCRIPTION = mMockEvents.getNextEvent().getDescription();
+    private static final String EMPTY = "";
+
+    @Rule
+    public final ActivityTestRule<MainActivity> mActivityRule =
+            new ActivityTestRule<>(MainActivity.class);
+
+    @Before
+    public void setup() {
+        onView(withText("Create"))
+                .inRoot(withDecorView(is(mActivityRule.getActivity().getWindow().getDecorView())))
+                .perform(click());
+    }
+
+    @Test
+    public void testCreateEventFragment() throws InterruptedException {
+        launchEventFragment();
+
+        // Now try with correct values
+        onView(withHint(is("Title"))).perform(
+                clearText(),
+                typeText(TITLE),
+                closeSoftKeyboard());
+
+        sleep(10000);
+
+        onView(withHint(is("Description"))).perform(
+                clearText(),
+                typeText(DESCRIPTION),
+                closeSoftKeyboard());
+
+        sleep(10000);
+
+        onView(withHint(is("Date"))).perform(
+                clearText(),
+                typeText(DATE),
+                closeSoftKeyboard());
+
+        sleep(10000);
+
+        onView(withId(R.id.createButton)).perform(
+                click());
```

Comment:
The problem comes from here, it thinks this inputs are empty and so doesn't trigger the correct functions. Only on Travis everything is fine on local. @dialexo @zyuiop

Reply:
Did you try with a local emulator with smaller screen size?

Reply:
There are also ways to launch travis containers locally, see https://stackoverflow.com/questions/21053657/how-to-run-travis-ci-locally/49019950#49019950 and https://github.com/travis-ci/worker, although I never tested them personally and don't know if they work